tough/src/cache.rs

```diff
@@ -101,6 +110,15 @@ impl<'a, T: Transport> Repository<'a, T> {
         }
     }
 
+    /// Prepends the version number to the role.json filename if using consistent snapshot mode.
+    fn delegated_filename(&self, name: &str) -> String {
+        if self.root.signed.consistent_snapshot {
+            format!("{}.{}.json", self.targets.signed.version, name)
```

Comment:
Using the targets role's version is incorrect, it should be the version of the delegated role itself. That can be found in the snapshot role.

Maybe this function needs to return `Option<String>`, then...

```suggestion
            format!("{}.{}.json", self.snapshot.signed.meta.get(format!("{}.json", name))?, name)
```

Might be better to always return `None` if the file isn't present in snapshot.json, though.